apps/authentication-service/src/shared/users/usersService.test.ts

```diff
@@ -0,0 +1,52 @@
+import { AUTH0_USER_BY_ID } from "@/test/handlers";
+import { exampleUserWithClientId } from "@/test/testData/users";
+import { server } from "@/test/testServer";
+import { http, HttpResponse } from "msw";
+import {
+  getUserClientId,
+  getUserById,
+  User,
+  setUserClientId,
+} from "./userService";
+
+describe("User Service tests", () => {
+  const USER_ID = "google-oauth2|115545703201865461059";
+  const CLIENT_ID: string = "test-client-id";
+
+  it("returns a user resource", async () => {
+    const user: User = await getUserById(USER_ID);
+    expect(user).not.toBe(undefined);
+    expect(user).toEqual(exampleUserWithClientId);
+  });
+
+  it("returns a user's client id", async () => {
+    const clientId = await getUserClientId(USER_ID);
+    expect(clientId).not.toBe(undefined);
+    expect(clientId).toBe(CLIENT_ID);
+  });
+
+  it("sets a user's client id", async () => {
+    const user = await setUserClientId(USER_ID, CLIENT_ID);
+    expect(user).not.toBe(undefined);
+    expect(user).toEqual(exampleUserWithClientId);
+  });
+
+  it("tries to set a client id for a user that already has one", async () => {
```

Comment:
I'd call this "handles an api failure" and make the error message more generic. This test as is isn't testing that it fails if the user already has a client. It's just testing that it handles errors from the api. 

My guess is that calling setUserClientId would actually just override the client if it were really called, so this test is giving us false confidence that we're handling something that we aren't.

Reply:
Okay. I see what you mean. This test is handling something that the usersService doesn't handle.

`createClient` (in `clientsService`)  is where `setUserClientId` is called. `createClient` does check the token user to see if they are already associated with a Client. If so, it won't create a new client, and won't set the client id on the user.